uvicorn/protocols/http/httptools_impl.py

```diff
@@ -137,7 +137,8 @@ def connection_lost(self, exc):
         if self.cycle and not self.cycle.response_complete:
             self.cycle.disconnected = True
         self.message_event.set()
-        self.flow.resume_writing()
+        if self.flow:
+            self.flow.resume_writing()
```

Comment:
I'm a bit confused what this resolves. It looks like it could only fail if `self.flow` is None, but it's not clear how that could possibly occur, since surely `connection_made` is always called prior to `connection lost`?

What's the traceback of the error that occurs?

Reply:
I'm not neccessarily against this (although I think we should use `if self.flow is not None`) since it'd be more strictly correct from a type-checked point of view. I'd just like to understand the issue better first.

Reply:
Here is the traceback on invocation of the above command:
```
ERROR: Exception in callback <bound method HttpToolsProtocol.connection_lost of <uvicorn.protocols.http.httptools_impl.HttpToolsProtocol object at 0x7fe275424710>>
handle: <Handle HttpToolsProtocol.connection_lost>
Traceback (most recent call last):
  File "uvloop/cbhandles.pyx", line 68, in uvloop.loop.Handle._run
  File "/usr/local/lib/python3.6/site-packages/uvicorn/protocols/http/httptools_impl.py", line 140, in connection_lost
    self.flow.resume_writing()
AttributeError: 'NoneType' object has no attribute 'resume_writing'
```

The `nc` call is not sending any traffic, just checking that the port is open.

Reply:
Okay, well let's use `if self.flow is not None` for the branch.